### PR TITLE
IRIS-849-csharp-fullpage

### PR DESCRIPTION
**Pull request by paweltomaszewskisaucelabs**
### Description
Add c# fullpage screenshot docs

### Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/visual-testing/integrations/csharp.md [hunk 1]
```diff
@@ -335,6 +336,33 @@ VisualClient = VisualClient.Create(Driver, Region.UsWest1, sauceUsername, sauceA
 VisualClient.CaptureDom = true;
 ```
 
+### Full page screenshots
+
+<FullPageDescription />
```
Comment (FriggaHel) [inline docs/visual-testing/integrations/csharp.md:341]: Not sure that's ideal:
- Casing is not the same
- Method names are not identical (`SauceVisualCheck` vs. `VisualCheck`)
- Not mentioning the `FullPageConfig` kind.

My first feeling when reading is that example and description does not match (which is the case actually).

Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/csharp.md]: fixed

File: docs/visual-testing/integrations/csharp.md [hunk 2]
```diff
@@ -335,6 +336,44 @@ VisualClient = VisualClient.Create(Driver, Region.UsWest1, sauceUsername, sauceA
...
+By default, only the current viewport is captured when `.VisualCheck` is used. You can opt in to capturing the entire page by using the `FullPage` option. It will capture everything by scrolling and stitching multiple screenshots together.
+Additionally, you have the option to configure full page settings using the `FullPageConfig` option.
+
+:::note
+It's recommended to use the `HideAfterFirstScroll` option for fixed or sticky position elements such as sticky headers or consent banners.
+:::
+
+Options:
+
+- `DelayAfterScrollMs`: Delay in ms after scrolling and before taking screenshots (helps with lazy loading content)
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:350]: What is the default value?

Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/csharp.md:350]: there is no default value so no value == no delay

Comment (FriggaHel) [reply docs/visual-testing/integrations/csharp.md:350]: Should we mention the default behaviour ?

Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/csharp.md:350]: IMO no need.
I would expect it to be 0 if nothing was mention in the docs. What about you guys?

File: docs/visual-testing/_partials/_fullpage-description.md [hunk 3]
```diff
@@ -6,5 +6,5 @@ It's recommended to use the `hideAfterFirstScroll` option for fixed or sticky po
 
 Options:
 
-- `delayAfterScrollMs`: Delay in ms after scrolling and before taking screenshots (helps with lazy loading content)
+- `delayAfterScrollMs`: Delay in ms after scrolling and before taking screenshots. Default value is 0 (helps with lazy loading content)
```
Comment (konraddysput) [inline docs/visual-testing/_partials/_fullpage-description.md:9]:
```suggestion
- `delayAfterScrollMs`: Delay in ms after scrolling and before taking screenshots. The default value is 0. We recommend using this option for lazy loading content.
```

File: docs/visual-testing/integrations/csharp.md [hunk 4]
```diff
@@ -333,6 +334,46 @@ VisualClient = VisualClient.Create(Driver, Region.UsWest1, sauceUsername, sauceA
 VisualClient.CaptureDom = true;
 ```
 
+### Full page screenshots
```
Comment (konraddysput) [inline docs/visual-testing/integrations/csharp.md:337]: There is no way to enable this option for every VisualCheck via Client options?

**Review by konraddysput — APPROVED** [review]
Can we please wait for Felix approval to land it?

File: docs/visual-testing/integrations/csharp.md (hunk 4 above)
Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/csharp.md:337]: no but we have task to do it
https://saucedev.atlassian.net/browse/IRIS-892

File: docs/visual-testing/integrations/java.md [hunk 5]
```diff
@@ -331,7 +331,17 @@ visual.sauceVisualCheck("Inventory Page", options);
 
 ### Full page screenshots
 
-<FullPageDescription />
+By default, only the current viewport is captured when `.sauceVisualCheck` is used. You can opt in to capturing the entire page by using the `enableFullPageScreenshots` option. It will capture everything by scrolling and stitching multiple screenshots together.
+Additionally, you have the option to configure full page settings using the `FullPageConfig` option.
+
+:::note
+It's recommended to use the `HideAfterFirstScroll` option for fixed or sticky position elements such as sticky headers or consent banners.
```
Comment (FriggaHel) [inline docs/visual-testing/integrations/java.md:338]: Is this correct for java ? Isn't it expected to use Builder ?

Casing looks off too.

File: docs/visual-testing/integrations/java.md [hunk 6]
```diff
@@ -331,7 +331,17 @@ visual.sauceVisualCheck("Inventory Page", options);
...
+:::
+
+Options:
+
+- `DelayAfterScrollMs`: Delay in ms after scrolling and before taking screenshots. The default value is 0. We recommend using this option for lazy loading content.
+- `HideAfterFirstScroll`: Hide elements on the page after first scroll (uses css selectors)
```
Comment (FriggaHel) [inline docs/visual-testing/integrations/java.md:344]: Same as previous comment.

File: docs/visual-testing/integrations/java.md
Comment (FriggaHel) [inline docs/visual-testing/integrations/java.md:1]: Is there a need for changes to that file ? Seems like `FullPageDescription` was doing exactly this.

Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/java.md:1]: If we want to provide specific description for java then yes we need to change it

File: docs/visual-testing/integrations/java.md (hunk 6 above)
Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/java.md]: fixed

File: docs/visual-testing/integrations/java.md (hunk 5 above)
Comment (paweltomaszewskisaucelabs) [reply docs/visual-testing/integrations/java.md]: fixed

**Review by FriggaHel — APPROVED** [review]

**Pull request merged**